src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -130,8 +134,9 @@ private AzureStorageDurabilityProvider GetAzureStorageStorageProvider(DurableCli
             AzureStorageOrchestrationServiceSettings settings = this.GetAzureStorageOrchestrationServiceSettings(connectionName, attribute.TaskHub);
 
             AzureStorageDurabilityProvider innerClient;
-            if (string.Equals(this.defaultSettings.TaskHubName, settings.TaskHubName, StringComparison.OrdinalIgnoreCase) &&
-                string.Equals(this.defaultSettings.StorageConnectionString, settings.StorageConnectionString, StringComparison.OrdinalIgnoreCase))
+            if (!attribute.ExternalClient &&
```

Comment:
I now remember that some customers use the ExternalClient attribute for external clients that use the DurableClient input binding, not just the factory. In that case, they technically don't need to provider a TaskHub name and/or storage connection string, as we just use the default values.

I think we are OK with not using the default storage provider in that case however, as the comment below really just wants us to use the defaultStorageProvider for clients that are running on the same instance as the triggers to bypass queue polling.

Reply:
Would it be better to go back to checking if `defaultStorageProvider` is not null instead of checking `attribute.ExternalClient`?

Reply:
in general, they should be synonymous, but it is probably more robust to check for the nullness of `defaultStorageProvider`. Let's consider moving it back to checking if `defaultStorageProvider` is null, with a comment explaining when it would be null and why it is ok to not return it in those scenarios.